frontend/src/scenes/insights/utils/dashboardTiles.ts

```diff
@@ -0,0 +1,28 @@
+import { DashboardTile, InsightModel } from '~/types'
+
+/**
+ * When changing an InsightModel there is some data that applied no matter where the insight is displayed or changed
+ * And other data that varies depending on whether the insight is on a dashboard and which dashboard it is on
+ * This is a convenience method to ensure that when receiving changes to an insight
+ * the correct dashboard specific data is kept
+ *
+ * returns an updated copy of the target
+ *
+ * @param target the InsightModel to update
+ * @param source the Insight data that is within a particular dashboard context
+ */
+export function mergeWithDashboardTile<T extends InsightModel | Partial<InsightModel>>(
+    target: T,
+    source: DashboardTile
+): T {
```

Comment:
If we're already nitting, then why `source` and `target`, not `insight` and `dashboardTile`? I also think the description could be a lot shorter, or just omitted, as here I think reading the code tells you with less time what's happening, than the comment.